server/src/main/java/org/elasticsearch/action/search/SearchQueryThenFetchAsyncAction.java

```diff
@@ -72,4 +89,67 @@ protected void onShardGroupFailure(int shardIndex, Exception exc) {
     protected SearchPhase getNextPhase(final SearchPhaseResults<SearchPhaseResult> results, final SearchPhaseContext context) {
         return new FetchSearchPhase(results, searchPhaseController, context);
     }
+
+    ShardSearchRequest buildShardSearchRequestInternal(SearchShardIterator shardIt) {
+        ShardSearchRequest request = super.buildShardSearchRequest(shardIt);
+        if (resultConsumer == null) {
+            return request;
+        }
+        List<TopFieldDocs> topDocsList = resultConsumer.getRemainingTopDocs().stream()
+            .filter(obj -> obj instanceof TopFieldDocs)
+            .map(obj -> (TopFieldDocs) obj)
+            .collect(Collectors.toList());
+        if (topDocsList.isEmpty()) {
+            return request;
+        }
+        SearchSourceBuilder source = request.source();
+        int trackTotalHits = source.trackTotalHitsUpTo() == null ? DEFAULT_TRACK_TOTAL_HITS_UP_TO : source.trackTotalHitsUpTo();
+        long totalHits = -1;
```

Comment:
It is not clear for me why a default value is `-1`?

Reply:
It should be 0, thanks. I pushed https://github.com/elastic/elasticsearch/pull/51852/commits/8eae737035abbf59b39194d364d68cb7a33d9fae